src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -415,13 +416,13 @@ void BindRangeType(NpgsqlTypeHandler elementHandler, PostgresRangeType pgRangeTy
 
             // It might be an unmapped enum/composite type, or some other unmapped type
             return (null, postgresType);
-
-            bool TryGetMapping(PostgresType pgType, out NpgsqlTypeMapping mapping)
-                => Mappings.TryGetValue(pgType.Name, out mapping) ||
-                   Mappings.TryGetValue(pgType.FullName, out mapping) ||
-                   pgType is PostgresDomainType domain && (
-                       Mappings.TryGetValue(domain.BaseType.Name, out mapping) ||
-                       Mappings.TryGetValue(domain.BaseType.FullName, out mapping));
         }
+
+        bool TryGetMapping(PostgresType pgType, [NotNullWhenTrue] out NpgsqlTypeMapping? mapping)
```

Comment:
Why move? You don't like local functions? :)

Reply:
It's a compiler limitation. It doesn't allow attributes on local functions :( Wrote a note to myself to raise an issue later or find an appropriate one later.

Reply:
Right, makes sense.

Reply:
dotnet/csharplang#1888 covers this limitation. Should be fixed in the next minor C# version.